modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/manager/v1_0/DefaultObjectEntryManagerImpl.java

```diff
@@ -209,16 +209,18 @@ public void deleteObjectEntry(
 
 	@Override
 	public ObjectEntry fetchObjectEntry(
-			DTOConverterContext dtoConverterContext,
```

Comment:
I think the ObjectDefinition here was added to improve performance because in some cases we already have the objectdefinition and if it was not passed then we needed to fetch it again from the objectEntryId as you made in your change. We need to analyze the uses of fetchObjectEntry and check it

I've made a quick search and it seems so, although I think we may have a bug because we need to check if the ObjectDefinition is null to fetch it only in this case.

Maybe because of this is better to continue with your breaking change and always fetch again the ObjectDefinition to avoid the risk of it being null.

We can talk about it if you want

Reply:
Mmmm, I like the part of improve performance. If we add a condition to check if the object definition is already there we can save a call to the DB and it is just a condition. Also, we avoid the breaking change and all the changes in other modules, like `object-dynamic-data-mapping` and `object-storage-salesforce`.

Reply:
In this case go ahead and change the code doing that and we avoid the breaking change.

Once you have changed it, you can forward the PR

Reply:
![](https://media.giphy.com/media/xTiIzsrT72FxfmWDsI/giphy.gif)